pkg/cluster/install.go

```diff
@@ -141,54 +142,100 @@ func (m *manager) Update(ctx context.Context) error {
-			steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.ensureGatewayCreate)),
-			steps.Action(m.createAPIServerPrivateEndpoint),
-			steps.Action(m.createCertificates),
-
-			// Run installer. For M5/M6 we will persist the graph inside the
-			// installer code since it's easier, but in the future, this data
-			// should be collected from Hive's outputs where needed.
-			steps.Action(m.callInstaller),
+func (m *manager) runHiveInstaller(ctx context.Context) error {
+	// TODO: Load from M6 database
+	installerPullspec, err := m.env.LiveConfig().DefaultInstallerPullSpec(ctx)
+	if err != nil {
+		return err
+	}
+
+	v := &api.OpenShiftVersion{
+		Version:           version.InstallStream.Version.String(),
+		OpenShiftPullspec: version.InstallStream.PullSpec,
+		InstallerPullspec: installerPullspec,
+	}
+
+	// Run installer. For M5/M6 we will persist the graph inside the installer
+	// code since it's easier, but in the future, this data should be collected
+	// from Hive's outputs where needed.
+	return m.hiveClusterManager.Install(ctx, m.subscriptionDoc, m.doc, v)
+}
+
+func (m *manager) bootstrap() []steps.Step {
+	s := []steps.Step{
+		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.validateResources)),
+		steps.Action(m.ensureACRToken),
+		steps.Action(m.ensureInfraID),
+		steps.Action(m.ensureSSHKey),
+		steps.Action(m.ensureStorageSuffix),
+		steps.Action(m.populateMTUSize),
+
+		steps.Action(m.createDNS),
+		steps.Action(m.initializeClusterSPClients), // must run before clusterSPObjectID
+		steps.Action(m.clusterSPObjectID),
+		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.ensureResourceGroup)),
+		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.enableServiceEndpoints)),
+		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.setMasterSubnetPolicies)),
+		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.deployStorageTemplate)),
+		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.attachNSGs)),
+		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.updateAPIIPEarly)),
+		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.createOrUpdateRouterIPEarly)),
+		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.ensureGatewayCreate)),
+		steps.Action(m.createAPIServerPrivateEndpoint),
+		steps.Action(m.createCertificates),
+
+		// We will always need a Hive namespace, whether we are installing
+		// via Hive or adopting
+		steps.Action(m.hiveCreateNamespace),
+	}
+
+	if m.installViaHive {
+		s = append(s,
+			steps.Action(m.hiveCreateNamespace),
+			steps.Action(m.runHiveInstaller),
+			// Give Hive 60 minutes to install the cluster, since this includes
+			// all of bootstrapping being complete
+			steps.Condition(m.hiveClusterInstallationComplete, 60*time.Minute, true),
+			steps.Condition(m.hiveClusterDeploymentReady, 5*time.Minute, true),
```

Comment:
The order of these is going to change in my follow-up PR, because of [the bug mentioned here](https://github.com/Azure/ARO-RP/pull/2215/files#r958581227).